pyccel/parser/syntactic.py

```diff
@@ -917,14 +917,14 @@ def _visit_Call(self, stmt):
         func = self._visit(stmt.func)
 
         if isinstance(func, Symbol):
-            f_name = func.name
-            if str(f_name) == "print":
+            f_name = str(func.name)
+            if f_name == "print":
                 func = PythonPrint(PythonTuple(*args))
             else:
-                func = Function(f_name)(*args)
+                func = FunctionCall(f_name, args)
```

Comment:
Could we not make the `Function.__call__` magic method return a `FunctionCall` object?

Reply:
It's not possible in the syntactic stage as `Function` is a sympy method but in the semantic stage we can make a `PyccelInternalFunction.__call__` and a `FunctionDef.__call__` return a `FunctionCall`

Reply:
So in the syntactic stage we create objects of type `sympy.Function`... Are we sure that we still need to do this?

Reply:
No. That is what is in the deleted line. `f_name` is a string at this line

Reply:
If we don't need `sympy.Function` anymore, then why can't we use our own `Function` here, and add it a `__call__` method?

Reply:
Sorry, I just realized that we have `FunctionDef` but not `Function` :sweat_smile:

Reply:
So a class:
```
class Function:
    def __init__(f_name):
        assert(isinstance(f_name, str))
        self._name = f_name
        
    def __call__(self, *args):
        return FunctionCall(self._name, args)
```
which is only used in the syntactic stage? I'm not sure I see what this adds. Presumably it slows things down though

Reply:
Yes, it's not useful if only used in the syntactic stage.